File: src/main/java/com/wafflestudio/draft/service/CourtService.java
package com.wafflestudio.draft.service;

import com.wafflestudio.draft.model.Court;
import com.wafflestudio.draft.repository.CourtRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

@Service
@Transactional(readOnly = true)
public class CourtService {
    @Autowired
    private CourtRepository courtRepository;

    public Optional<Court> getCourtById(Long id) {
        return courtRepository.findById(id);
    }
}

File: src/main/kotlin/com/wafflestudio/draft/service/CourtService.kt
package com.wafflestudio.draft.service

import com.wafflestudio.draft.error.CourtNotFoundException
import com.wafflestudio.draft.model.Court
import com.wafflestudio.draft.repository.CourtRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class CourtService(private val courtRepository: CourtRepository) {
    fun getCourtById(id: Long): Court {
        return courtRepository.findById(id).orElseThrow(::CourtNotFoundException)!!
    }

    fun findCourtsByName(name: String?): List<Court> {
        return courtRepository.findByNameContaining(name)
    }

    fun findCourtsByRegionId(id: Long): List<Court>? {
        return courtRepository.findAllByRegion_Id(id)
    }
}
